### PR TITLE
Changes for showing recording SAS uri on recording stopped ribbon.

### PR DESCRIPTION
**Pull request by ravithanneeru**
## Purpose

1. Show recording SAS Uri once recording stopped.
2. Formatting using prettier.
* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: Calling/BlobStorageHelper.cs [hunk 1]
```diff
@@ -93,5 +100,32 @@ public static async Task<BlobStorageHelperInfo> UploadFileAsync(string connectio
...
+        /// <param name="blobClient">A reference to the blobclient object.</param>
+        /// <returns>A string containing the URI for the blob, with the SAS token appended.</returns>
+        static string GetBlobSasUri(string containerName, BlobClient blobClient)
+        {
+            // Create a SAS token that's valid for a given time.
+            BlobSasBuilder sasBuilder = new BlobSasBuilder()
+            {
+                BlobContainerName = containerName,
+                BlobName = blobClient.Name,
+                Resource = "b"
```
Comment (alkwa-msft) [inline Calling/BlobStorageHelper.cs:118]: Can this be anything? I want to make sure we are setting our customers up for success and if we just set a random string it seems confusing?

File: Calling/ClientApp/src/Utils/Utils.ts [hunk 2]
```diff
@@ -77,6 +77,20 @@ export const utils = {
...
     }
   },
+  getRecordingLink: async (id: string): Promise<string> => {
+    try {
+      const response = await fetch('/recording/getRecordingLink?serverCallId=' + id);
+      if (response.ok) {
+        const content = await response.json();
+        return content.uri;
+      } else {
+        return '';
```
Comment (alkwa-msft) [inline Calling/ClientApp/src/Utils/Utils.ts:87]: what is the best practices here if we can't get a response from the server?

File: Calling/ClientApp/src/Utils/Utils.ts [hunk 3]
```diff
@@ -77,6 +77,20 @@ export const utils = {
...
+      }
+    } catch (e) {
+      console.error('Recording is not available for download at this moment. Please try again later.');
+      return '';
```
Comment (alkwa-msft) [inline Calling/ClientApp/src/Utils/Utils.ts:91]: Perhaps we can use the error response above to help customers in the case they can't download their recording?

File: Calling/BlobStorageHelper.cs (hunk 1 above)
Comment (ravithanneeru) [reply Calling/BlobStorageHelper.cs:118]: No, it's not anything. If the shared resource is of type blob, we should give resource attribute value as "b". Reference https://docs.microsoft.com/en-us/dotnet/api/azure.storage.sas.blobsasbuilder.resource?view=azure-dotnet#Azure_Storage_Sas_BlobSasBuilder_Resource

File: Calling/ClientApp/src/Utils/Utils.ts (hunk 2 above)
Comment (PorterNan) [reply Calling/ClientApp/src/Utils/Utils.ts:87]: I believe we could use response code from server side, and then add our own error message in the response

File: Calling/ClientApp/src/components/ComplianceBanner.tsx [hunk 4]
```diff
@@ -1,180 +1,196 @@
...
-    const RECORDING_STOPPED = 6;
-    const RECORDING_STARTED = 7;
-    const RECORDING_STOPPED_AND_SAVED = 8;
-    const NO_STATE = 0;
+  const [previousCallTranscribeState, setPreviousCallTranscribeState] = useState(false);
+  const [previousCallRecordState, setPreviousCallRecordState] = useState(false);
+  const [variant, setVariant] = useState(0);
 
-    const {callTranscribeState, callRecordState} = props;
+  const TRANSCRIPTION_STOPPED_STILL_RECORDING = 1;
```
Comment (PorterNan) [inline Calling/ClientApp/src/components/ComplianceBanner.tsx:16]: Let's move these into constants.ts

File: Calling/BlobStorageHelper.cs [hunk 5]
```diff
@@ -84,6 +85,12 @@ public static async Task<BlobStorageHelperInfo> UploadFileAsync(string connectio
                 FileStream uploadFileStream = File.OpenRead(filePath);
                 BlobContentInfo status = await blobClient.UploadAsync(uploadFileStream, true);
                 uploadFileStream.Close();
+
+                if(filePath.EndsWith("mp4"))
```
Comment (PorterNan) [inline Calling/BlobStorageHelper.cs:89]: Is mp4 the only format it will generate?

**Review by PorterNan — APPROVED** [review]

File: Calling/BlobStorageHelper.cs (hunk 5 above)
Comment (ravithanneeru) [reply Calling/BlobStorageHelper.cs:89]: At the moment, we are downloading and uploading recording in mp4 format only. There can be other formats which we didn't explored.

**Review by alkwa-msft — APPROVED** [review]

**Pull request merged**